packages/mobile/src/analytics/ValoraAnalytics.ts

```diff
@@ -0,0 +1,150 @@
+import Analytics, { Analytics as analytics } from '@segment/analytics-react-native'
+import Firebase from '@segment/analytics-react-native-firebase'
+import DeviceInfo from 'react-native-device-info'
+import { AppEvents } from 'src/analytics/Events'
+import { AnalyticsPropertiesList } from 'src/analytics/Properties'
+import { SEGMENT_API_KEY } from 'src/config'
+import { store } from 'src/redux/store'
+import Logger from 'src/utils/Logger'
+import * as Web3Utils from 'web3-utils'
+
+const TAG = 'ValoraAnalytics'
+
+async function getDeviceInfo() {
+  return {
+    AppName: DeviceInfo.getApplicationName(),
+    Brand: DeviceInfo.getBrand(),
+    BuildNumber: DeviceInfo.getBuildNumber(),
+    BundleId: DeviceInfo.getBundleId(),
+    Carrier: await DeviceInfo.getCarrier(),
+    DeviceId: DeviceInfo.getDeviceId(),
+    DeviceName: await DeviceInfo.getDeviceName(), // NOTE(nitya) this might contain PII, monitor
+    FirstInstallTime: await DeviceInfo.getFirstInstallTime(),
+    FontScale: await DeviceInfo.getFontScale(),
+    FreeDiskStorage: await DeviceInfo.getFreeDiskStorage(),
+    InstallReferrer: await DeviceInfo.getInstallReferrer(),
+    InstanceID: await DeviceInfo.getInstanceId(),
+    LastUpdateTime: await DeviceInfo.getLastUpdateTime(),
+    Manufacturer: await DeviceInfo.getManufacturer(),
+    MaxMemory: await DeviceInfo.getMaxMemory(),
+    Model: DeviceInfo.getModel(),
+    ReadableVersion: DeviceInfo.getReadableVersion(),
+    SerialNumber: await DeviceInfo.getSerialNumber(),
+    SystemName: DeviceInfo.getSystemName(),
+    SystemVersion: DeviceInfo.getSystemVersion(),
+    TotalDiskCapacity: await DeviceInfo.getTotalDiskCapacity(),
+    TotalMemory: await DeviceInfo.getTotalMemory(),
+    UniqueID: DeviceInfo.getUniqueId(),
+    UserAgent: await DeviceInfo.getUserAgent(),
+    Version: DeviceInfo.getVersion(),
+    isEmulator: await DeviceInfo.isEmulator(),
+    isTablet: DeviceInfo.isTablet(),
+  }
+}
+
+const SEGMENT_OPTIONS: analytics.Configuration = {
+  using: [Firebase],
+  flushAt: 20,
+  trackAttributionData: false,
+  debug: __DEV__,
+  trackAppLifecycleEvents: true,
+  recordScreenViews: true,
+  ios: {
+    trackAdvertising: false,
+    trackDeepLinks: true,
+  },
+}
+
+class ValoraAnalytics {
+  sessionId: string
+  userAddress: string | null | undefined
+  deviceInfo: object
+
+  constructor() {
+    this.sessionId = ''
+    this.deviceInfo = {}
+    this.userAddress = 'not yet set'
```

Comment:
@nityas how would you feel about leaving out the user address. Given the growing scope of analytics, I'm not sure how ethical it is to tie all this activity to particular addresses

Reply:
hm...on ethics I lean on the ability of any user to opt out of analytics at any time, which is far more than most consumer apps offer. In general, we could always tie together events implicitly through anonymous IDs, so I see this as a way to make sure our data is accurate (this also sets us up better to accurately bulk delete user data if a request is made, which is a feature I could see us adding in the future)

Reply:
Are there any specific data points you're hesitant to tie to address or is this more of a general concern?

I think no collection of PII, an easily accessible opt-out of all analytics, and clear documentation on what's being collected sets a sufficiently high bar.

Reply:
@jmrossy because we're tracking transaction hashes, we can identify the user address from the hash sent from a particular wallet. I agree that we should not log any PII (phone number, name, etc) but would argue that the address falls more in the category of on-chain data that is acceptable to track

Reply:
Yeah this is a tricky one. On the one hand, all on-chain activity is already publicly tied to an address. On the other, addresses are the blockchain-equivalent of user IDs so I feel like including it weakens our claim that we do analytics in an anonymous way. 

It certainly helps us make correlations across services (like identifying misbehaving PGPNP users). 

I have an inherent bias towards logging less, tracking less, but I'll defer to you three on this one :)